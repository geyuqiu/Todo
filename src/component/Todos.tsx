import React, {useState} from "react";

import {TodoTable} from './TodoTable/TodoTable';
import {useFetch} from '../hooks/useFetch';

export const Todos = () => {
	const [isLoading, setLoading] = useState(true);

	const {todos} = useFetch(setLoading);

	return (
		<>
			<TodoTable todos={todos} isLoading={isLoading}/>
		</>
	);
};
