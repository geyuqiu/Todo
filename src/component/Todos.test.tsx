import {render} from "@testing-library/react";
import React from "react";

import {Todos} from "./Todos";

describe("WalletView", () => {
	it("should match snapshot", () => {
		const { container } = render(<Todos />);
		expect(container).toMatchSnapshot();
	});
});
