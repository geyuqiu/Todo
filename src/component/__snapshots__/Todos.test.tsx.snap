// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletView should match snapshot 1`] = `
<div>
  <div
    class="ml-6 mr-3 sm:mx-8 pt-12"
  >
    <h1
      class="font-bold mb-3"
    >
      Todos
    </h1>
    <div
      class="flex sm:flex-row flex-col mb-6"
    >
      <div
        class="flex justify-between mb-3 sm:mb-0 items-center w-full"
      >
        <label>
          Search: 
        </label>
        <input
          class="w-full ml-3 p-2 border-gray-dark border"
          placeholder="keyword..."
          value=""
        />
      </div>
      <div
        class="flex justify-between items-center w-full sm:ml-3"
      >
        <label>
          Completed: 
        </label>
        <select
          class="w-full ml-3 p-2 border-gray-dark bg-white border"
        >
          <option
            value=""
          >
            -
          </option>
          <option
            value="yes"
          >
            yes
          </option>
          <option
            value="no"
          >
            no
          </option>
        </select>
      </div>
    </div>
    <p
      data-testid="Table__Loading"
    >
      Loading ...
    </p>
  </div>
</div>
`;
